meshtastic/interfaces/ble/discovery.py

```
@@ -524,7 +562,21 @@ def _discard_cached_client() -> None:
+                # Compatibility for minimal test doubles that still expose
+                # underscore-prefixed discover helpers.
+                discover = getattr(client, "_discover", None)
+            if not callable(discover) or _is_unconfigured_mock_callable(discover):
+                with self._client_lock:
+                    if self._client is client:
+                        self._client = None
+                raise DiscoveryClientError.invalid_client(
+                    resolved_factory,
+                    type(client),
+                    ["discover", "_discover"],
+                )
```

Ruff (0.15.5)

[warning] Abstract `raise` to an inner function

(TRY301)